File: Search/Find_Minimum_in_Rotated_Sorted_Array.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0, right = nums.length - 1;
        int middle;

        while (left <= right) {
            if (nums[left] <= nums[right]) {
                return nums[left];
            }

            middle = (right + left) / 2;
            if (nums[middle] >= nums[right]) {
                left = middle + 1;
            } else {
                right = middle;
            }
        }

        return 0;
    }
}

File: Search/Find_Minimum_in_Rotated_Sorted_Array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left, right, minimum = 0, len(nums) - 1, float("inf")

        while left <= right:
            mid = (right + left) // 2
            minimum = min(minimum, nums[mid])

            if nums[mid] > nums[right]:
                left = mid + 1
            else:
                right = mid - 1
        
        return minimum